build/dsc.rake

```diff
@@ -60,7 +60,36 @@ eod
         cmd += "git checkout #{ENV['DSC_REF']} &&" if ENV['DSC_REF']
         cmd += "git submodule update --init --recursive"
 
-        sh cmd
+        sh cmd unless File.exists?(dsc_resources_path_tmp)
+
+        puts "Creating sha file"
+        File.open("#{default_dsc_module_path}/dsc_resource_release_tags.md", 'w+') do |file|
+          file.write("## SHA tags for each DSC Resource repo\n")
+          file.write("\n")
+          file.write("DSC Resource | SHA | Release Tag\n")
+          file.write("------------ | --- | -----\n")
+          
+          Dir["#{dsc_resources_path_tmp}/xDSCResources/*"].each do |path|
+            if File.directory?(path)
+              puts "Finding latest tag for #{path}"
+              FileUtils.cd(path) do
+                begin
+                  repo = Git.open(path, :log => Logger.new(STDOUT))
```

Comment:
Don't think you need to pull the `git` gem in for this... just use git at the command line. For instance:

```
[puppet-agent] git rev-parse --short HEAD                   10:34:45  ☁  master ☂ ⚡ ✭
183c2da
[puppet-agent] git describe --tags --abbrev=0               10:35:00  ☁  master ☂ ⚡ ✭
1.4.1
[puppet-agent] git describe                                 10:35:07  ☁  master ☂ ⚡ ✭
1.4.1-224-g183c2da
```